pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -187,6 +199,48 @@ func allShardsServingWrites(shardNames []string, shardStatus map[string]planetsc
 	return result
 }
 
+func totalReadytablets(shardNames []string, shardStatus map[string]planetscalev2.VitessKeyspaceShardStatus) int32 {
+	// A partitioning with no shards has no ready tablets.
+	if len(shardNames) == 0 {
+		return 0
+	}
```

Comment:
This doesn't need to be special case. The loop will just iterate 0 times.

Reply:
Removed from all.